src/Contexts/Shared/infrastructure/QueryBus/QueryHandlersInformation.ts

```diff
@@ -0,0 +1,32 @@
+import { Query } from '../../domain/Query';
+import { QueryHandler } from '../../domain/QueryHandler';
+import { Response } from '../../domain/Response';
+import { QueryNotRegisteredError } from '../../domain/QueryNotRegisteredError';
+
+export class QueryHandlersInformation {
+  private queryHandlersMap: Map<Query, QueryHandler<Query, Response>>;
+
+  constructor(queryHandlers: Array<QueryHandler<Query, Response>>) {
+    this.queryHandlersMap = this.formatHandlers(queryHandlers);
+  }
+
+  private formatHandlers(queryHandlers: Array<QueryHandler<Query, Response>>): Map<Query, QueryHandler<Query, Response>> {
+    const handlersMap = new Map();
+
+    queryHandlers.forEach(queryHandler => {
+      handlersMap.set(queryHandler.subscribedTo(), queryHandler);
+    });
+
+    return handlersMap;
+  }
+
+  public search(query: Query): QueryHandler<Query, Response> {
+    const queryHandler = this.queryHandlersMap.get(query.constructor);
+
+    if (!queryHandler) {
+      throw new QueryNotRegisteredError(query);
+    }
```

Comment:
Are we handling these exception somewhere?

Reply:
This exception and the CommandNotRegisteredError are handled by the controller, throwing a 500 error. Is not handled as a specific error but its a general error.

https://github.com/CodelyTV/typescript-ddd-skeleton/blob/d6b7729e96f7ef64681101151e794f38913ed9d9/src/apps/mooc_backend/controllers/CoursesCounterGetController.ts#L21